### ISSUE TITLE
ob-nav-tree - Cannot add aria-label to menus

### ISSUE THREAD

**Issue by shakeeldoushantimrith** [oblique-bit/oblique#18]
Hello,

We are currently creating an application where visually impaired users will be using the application using a screen reader.
As such, we were asked to add aria-label attributes in the application.

For the side menu, we are using the ob-nav-tree and the ObNavTreeItemModel for each menu.

Can you please guide us on how to add the aria-label attributes to those menus?

Thanks.

Kind Regards,
Shakeel

**Comment by gillerr** [comment]
Hello

I don't understand why you want to add `aria-label`s. Since the link itself will already be read out loud by a screen reader, there's no need for an additional `aria-label`, or am I missing something? Or do you want to add a label on the `ul` element itself?

Anyway, this component is currently being improved for accessibility, so this is the right moment to clarify this.

Could you explain how and why you would like to add `aria-label`?

**Comment by shakeeldoushantimrith** [comment]
Hello,

The application is currently being used in production and one of the problem they faced is that the word "Jump View" is read out when navigating on the different menu item and a proper description is missing.
They mentioned that "A description of the function of the menu item would be necessary so that a person with visual impairment receives all the necessary information".

I guess they are looking for an aria-description property.

Can you please guide us on how to proceed with adding this?

Thanking you in advance.

Kind Regards,
Shakeel

**Comment by gillerr** [comment]
Ok, if I understand correctly, you need the possibility to add an `aria-labelledby` on  the `ul` element of the the `ob-nav-tree` . This feature will be integrated in Oblique 6.1 due this month. In fact, the `ob-nav-tree` will be refactored according to the accessibility guidelines of the W3C: https://www.w3.org/TR/wai-aria-practices-1.1/examples/treeview/treeview-2/treeview-2a.html

Will that solve your problem?

Cheers
René

**Comment by shakeeldoushantimrith** [comment]
Hello,

Yes, the presence of this attribute will solve this issue.

Thanks.

Regards,
Shakeel

**Comment by gillerr** [comment]
`treeAriaLabelledBy` and  `treeAriaLabel` inputs have been added to the `ob-nav-tree`. Both attributes will be set on the `ul` element.
Released with Oblique 6.1.0
